gensim/matutils.py

```diff
@@ -332,9 +343,15 @@ def unitvec(vec):
     Output will be in the same format as input (i.e., gensim vector=>gensim vector,
     or numpy array=>numpy array, scipy.sparse=>scipy.sparse).
     """
+    norm = isnorm(norm)
     if scipy.sparse.issparse(vec): # convert scipy.sparse to standard numpy array
         vec = vec.tocsr()
-        veclen = numpy.sqrt(numpy.sum(vec.data ** 2))
+        if norm == 'l0':
+            veclen = len(vec.data)
```

Comment:
Yes true.... Could you please elaborate more on the scaling part? Will it be affecting efficiency since as the corpus size increases we cannot use the fact that `f(ax) = (a^k)f(x)`? Should I remove the support for zero "norm"?